tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -111,13 +111,8 @@ public static void main(String[] args) throws Exception {
                 producer.initTransactions();
 
             /* setup perf test */
-            byte[] payload = null;
+            byte[] payload = new byte[recordSize];
```

Comment:
There is a null check for `recordSize` so is it (`new byte[recordSize]`) a potential NPE ?

Reply:
Got, move it to inside loop.

Reply:
@lamberken thanks for updating code. This change makes it create new array for each loop (as @tombentley mentioned). Could you avoid this performance regression? For example, we can initialize the `payload` before loop and then assign the value to each byte only if `payloadFilePath == null && recordSize != null`.

Reply:
@chia7712 patch updated. But I don't agree it can cause serious gc.

Reply:
@lamberken thanks for updating code.

> But I don't agree it can cause serious gc.

You are right and it is NOT a bad thing to do this "small" optimization, right? :)